codingstandards/listproperties.md

```diff
@@ -0,0 +1,128 @@
+# List Properties
+
+Many types, especially data structures, will contain properties of a list type,
+such as `IList<T>` or `IReadOnlyList<T>`.  List properties should be implemented
+under the following guidelines, especially `public` properties.
+
+- The list property itself should be read-only.
+- The list property should be initialized to a new instance when the containing data structure is constructed.
+- The list property should be of an interface type when possible.  That is, use `IList<T>` rather than `List<T>`.
+- Use the read-only collections when appropriate, such as `IReadOnlyCollection<T>`, `IReadOnlyList<T>`, and `IReadOnlyDictionary<TKey,TValue>`.
+
+## Validation
+
+As with any type, validation of the input should be performed as early as possible.
+In most cases, this can be accomplished in the constructor of the class.
+In the cases of lists, however, when a list is expected to only contain certain values,
+validation needs to occur as values are added, inserted, or set using an indexer.
+
+Here is an example of how to **not** validate the entries of a list.
+Note the list property, `Attributes`, correctly defined as a read-only property as `IList<string>`.
+
+```csharp
+public class Example
+{
+   public string Property { get; }
+
+   public IList<string> Attributes { get; } = new List<string>();
+
+   public Example(string property)
+   {
+      Property = property;
+   }
+}
+```
+
+Lets say the `Attributes` list should not contain any null or empty values.
+Before performing an operation, we may want to validate the `Attributes` property
```

Comment:
@Izzmo I did put in an intentional line break there.  I don't like really long lines in my text editor when I'm authoring it, causing me to use horizontal scrolling.  Same with HTML, I rarely let my lines get too long.  If possible, though, I do try to break them on the sentence.

You did have me concerned, though.  I thought maybe github's markdown honored single line breaks, but quickly verified that's not the case.

Is there a GTM/industry standard, here, @Izzmo that I'm violating?